File: r/feature_scaling.r
# Import dataset
df = read.csv('data.csv')

# caTools contains several basic utility functions including: moving 
# (rolling, running) window statistic functions, read/write for GIF 
# and ENVI binary files, fast calculation of AUC, LogitBoost classifier, 
# base64 encoder/decoder, round-off-error-free sum and cumsum, etc.
install.packages('caTools')
library(caTools)

# Create Boolean vector with random split
# SplitRatio specifies proportion of training set
split = sample.split(df$y, SplitRatio=0.8)

# Split into training and test set
train = subset(df, split == TRUE)
test = subset(df, split == FALSE)

# Feature scaling
# Note that features that are factors cannot be scaled. Therefore,
# we must only pick the columns that are numerical data.
train[, 2:3] = scale(train[, 2:3])
test[, 2:3] = scale(test[, 2:3])

File: python/feature_scaling.py
import pandas as pd

# Import dataset
df = pd.read_csv('data.csv')
X = df.iloc[:, :-1].values
y = df.iloc[:, -1].values

# Split into training and test set
# test_size specifies proportion of the test set
from sklearn.cross_validation import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2)

# Feature scaling
from sklearn.preprocessing import StandardScaler
sc_X = StandardScaler()
X_train = sc_X.fit_transform(X_train)
X_test = sc_X.transform(X_test)